Enable supplying custom SQL to be run as part of initialization

mogul opened the pull request:
This enables any DBA to easily go to town customizing PostgREST's behavior by customizing the DB content.

I'm at a bit of a loss as to what default `.sql` file we can provide here that will demonstrate doing something useful while not being overly prescriptive. Got any ideas?

mogul committed:
Enable supplying custom SQL to be run as part of initialization

mheadd committed:
Add sample sql file to make tables read only

mheadd commented:
How about [a simple SQL script](https://github.com/GSA/cloudgov-demo-postgrest/pull/4/commits/69fce9c457aa51f612f42c9de2aca8eff6722064) to make the tables in the database read only? This way, if someone wanted to only allow read access to the data, and not allow updates / insertions, something like this might do it.

mheadd committed:
Update sample sql file to make tables read only

mheadd commented on data/read-only.sql:
```diff
@@ -0,0 +1,2 @@
+-- Sample SQL file to make tables read only
+REVOKE INSERT, UPDATE, DELETE, TRUNCATE ON ALL TABLES IN SCHEMA public FROM anon;
```
@mogul I don't think this SQL script will work as intended without the anonymous role being set via the `vars.yml` file. So the anonymous role in that file would be the one referenced here.

mheadd reviewed (commented):
I added a simple SQL file to alter all tables to read only for the anonymous role. It's not working quite yet though, as I can't get the anonymous role set via the `vars.yml` file. If this doesn't seem like a good approach to a generic SQL script to demonstrate the functionality, feel free to ditch it.

mogul committed:
Go back to bringing in the db_anon_role from vars.yml

mogul committed:
Change default to a new schema, and do our best to make sure it exists

The Python script is always run now.

mogul committed:
Use `with` to scope commit-worthy transactions

mogul committed:
Refactor data initialization to be driven by the Python script

This enables us to first setup the anonymous role, then sequence import of CSV and SQL files.

mogul committed:
Use medium-psql plan, and improve error-handling for the deploy script

Note the change to medium-psql, which enables setting up roles in the DB!
This is necessary for now to ensure that the DB doesn't expose all the tables in the public schema
as writeable. However, this kind of DB can take several minutes to come up! So there's now a loop
in here to wait for that to complete.

A commit to come will remove the parameterization of the anonymous role so that `data/*.sql` files
can be explicit when referring to roles to grant/revoke.

mheadd reviewed (approved):
Perfect! I ran the set up and then tested to make sure the newly created table is read only.

mogul commented:
It works, but I don't like that (1) it takes like five minutes to spin up a `medium-psql` just to get `CREATEROLE` capabilities, and (2) the role is configurable even though `.sql` files are static and can't really incorporate changes.

I'm investigating (1) [upstream with the cloud.gov team here](https://gsa-tts.slack.com/archives/C09CR1Q9Z/p1583784617088300?thread_ts=1583784296.087400&cid=C09CR1Q9Z) and I'll hard-code the `anonymous` role name in my next commit.

mogul commented:
I found a way to set "variables" in the session before calling all the .sql files, so I'll make that change. Gotta go back to putting time into my "day job" for a while though! ;)

mogul commented:
Just so's I can close the tab: When I come back to this I want to make the setup of roles more idempotent, and [this SO question](https://stackoverflow.com/questions/8092086/create-postgresql-role-user-if-it-doesnt-exist) is relevant for that.
